crates/rattler_repodata_gateway/src/fetch/mod.rs

```diff
@@ -613,7 +648,7 @@ async fn stream_and_decode_to_file(
     // Decode, hash and write the data to the file.
     let bytes = tokio::io::copy(&mut decoded_repo_data_json_bytes, &mut hashing_file_writer)
         .await
-        .map_err(FetchRepoDataError::FailedToDownloadRepoData)?;
+        .map_err(RepoDataNotFoundError::FileSystemError)?;
 
```

Comment:
This is also a proper error. Not a not-found error.

Reply:
Changed back